src/dials/algorithms/integration/integrator.py

```diff
@@ -1235,7 +1235,7 @@ def _determine_max_memory_needed(experiments, reflections):
                         if experiment.imageset == imageset:
                             subset.extend(reflections.select(reflections["id"] == j))
                 try:
-                    if imageset.get_scan():
+                    if imageset.get_scan() and not imageset.get_scan().is_still():
                         frame0, frame1 = imageset.get_scan().get_array_range()
```

Comment:
```suggestion
                    if imageset.get_scan():
                        if not imageset.get_scan().is_still():
                            frame0, frame1 = imageset.get_scan().get_array_range()
                        else:
                            frame0, frame1 = (0, len(imageset))
                    else:
                        frame0, frame1 = (0, len(imageset))
```
When testing, I found that this line raised a `'NoneType' object has no attribute 'is_still'` error (my process was import, find_spots, ssx_index, integrate), so I think this should be guarded against in this function. It looks like the current logic assumes that the first call to `imageset.get_scan()` will raise a RuntimeError, but I don't think that is always the case anymore.
In terms of the effect on ImageSequences, we don't usually have an ImageSequence at this point, as not all images are indexable so the sequence has already been split into single-image imagesets (and I don't usually trigger this code path anyway). So I'm happy that this is a safe change for current workflows.

Reply:
Hi, thanks for the catch.  I've taken your proposal and modified it slightly and checked that in. It now removes the try/except block and does all the checks explicitly.  Can you review/test this version?  Thanks!

Reply:
Thanks, confirmed that your modification works, and is much cleaner code 🙂